cmux.go

```diff
@@ -177,7 +177,8 @@ func (l muxListener) Accept() (net.Conn, error) {
 // MuxConn wraps a net.Conn and provides transparent sniffing of connection data.
 type MuxConn struct {
 	net.Conn
-	buf buffer
+	buf     bytes.Buffer
```

Comment:
you can't possibly hide `buf` in the sniffer because MuxConn needs to read from it after all the sniffing is complete.